### PR TITLE
FSR: only set in Gamescope (remove Wine FSR)

### PR DESCRIPTION
**Pull request by SuperSamus**
# Description
FSR settings are moved in Gamescope. Wine FSR isn't set anymore.
Motivation: #2755

TL;DR: Right now, FSR is enabled in two places:
- Wine-GE, that is, only a specific version of Wine. It also already enables it by default, and doesn't even recommend to disable it (#2636).
- Gamescope, which obviously takes effect only if Gamescope is enabled.

First, having two separate FSR "enablers" is messy, as you need logic to avoid enabling both of them at the same time (and Bottles isn't doing that). Considering the options, I opted to simply remove Wine FSR.
Second, the conditions mentioned above aren't communicated at all by having FSR as a standalone setting, which is why I opted to move them into Gamescope's page.

Some comments on what's missing:
- Maybe it should communicate somewhere that Wine-GE already enables FSR, thus if you use it, you shouldn't enable it in Gamescope (still, it wasn't communicated even before).
- Didn't touch .po files yet.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?
Built Bottles myself, tested if the UI was correct, launched a game.

### CODE REVIEW

**Comment by SuperSamus** [comment]
(I'm waiting for comments before doing anything else)

**Comment by TheEvilSkeleton** [comment]
No worries. We've been busy for a few months now, so we're taking our time. I'm in a trip

**Comment by TheEvilSkeleton** [comment]
Can you resolve the conflicts? I did it locally, but I can't figure out how to push it directly to the MR.

**Comment by TheEvilSkeleton** [comment]
Thanks

File: bottles/frontend/ui/dialog-gamescope.blp [hunk 1]
```diff
@@ -179,6 +179,36 @@ template GamescopeDialog : .AdwWindow {
...
+            Adjustment spin_sharpening_strength {
+              step-increment: 4;
+              upper: 20;
+            }
+
+            ;
+          }
+        }
+      }
+
```
Comment (TheEvilSkeleton) [inline bottles/frontend/ui/dialog-gamescope.blp:211]: In my opinion, it makes more sense to move these above "Miscellaneous", since it's normal to put "Others" sections in the end.

I'm not too fond of the interface. I'm experimenting locally, but I'll see if we could come up with something better.

File: bottles/frontend/windows/gamescope.py [hunk 2]
```diff
@@ -32,9 +32,10 @@ class GamescopeDialog(Adw.Window):
     switch_scaling = Gtk.Template.Child()
     toggle_borderless = Gtk.Template.Child()
     toggle_fullscreen = Gtk.Template.Child()
+    switch_fsr = Gtk.Template.Child()
+    spin_sharpening_strength = Gtk.Template.Child()
     btn_save = Gtk.Template.Child()
     btn_cancel = Gtk.Template.Child()
```
Comment (TheEvilSkeleton) [inline bottles/frontend/windows/gamescope.py:38]: We shouldn't modify 'unrelated' lines for `git blame`.

```suggestion
    btn_cancel = Gtk.Template.Child()

```

**Review by TheEvilSkeleton — CHANGES_REQUESTED** [review]

File: bottles/backend/models/config.py [hunk 3]
```diff
@@ -88,8 +88,7 @@ class BottleParams(DictCompatMixIn):
     gamescope_fullscreen: bool = True
     sync: str = "wine"
     fsr: bool = False
-    fsr_sharpening_strength: int = 2
-    fsr_quality_mode: str = "none"
+    fsr_sharpening_strength: int = 8
```
[the remark this answers is not in the document]
Comment (SuperSamus) [reply bottles/backend/models/config.py:91]: If you look at other deleted lines you'll see that Wine's FSR and Gamescope's FSR have different ranges (0-5 and 0-20).
Bottles used Wine's range, and multiplied it by 4 when using FSR on Gamescope.
Let's simply use Gamescope's range directly.

[the remark this answers is not in the document]
Comment (TheEvilSkeleton) [reply bottles/backend/models/config.py:91]: Ah, thanks for the explanation :+1:

File: bottles/frontend/ui/dialog-gamescope.blp (hunk 1 above)
Comment (TheEvilSkeleton) [reply bottles/frontend/ui/dialog-gamescope.blp:211]: Yeah, I can't think of a better design that fits with the current dialog. In this case, can you move the FSR section above Miscellaneous? I'll just merge it afterwards.

**Comment by SuperSamus** [comment]
One last note. Logs now say:
```
(WARNING) Skipping unexpected config 'fsr_quality_mode' in BottleParams
```

(This config is, in fact, not used anymore.)
Is this log something important to deal with, or not?

**Comment by koplo199** [comment]
Note:
- Since `Wine-GE-Proton8-11`/`GE-Proton8-7` the bug caused by disabling FSR has been [fixed](https://github.com/GloriousEggroll/proton-ge-custom/releases/tag/GE-Proton8-7)
- Since `Wine-GE-Proton8-14`/`GE-Proton8-12` FSR is [disabled by default](https://github.com/GloriousEggroll/proton-ge-custom/releases/tag/GE-Proton8-12)

It adds to the fact that not communicating about `Wine-GE`/`GE-Proton` enabling FSR (between specific, older versions that is) maybe is just fine.

**Comment by orowith2os** [comment]
I think it would be better if we only dealed with gamescope FSR, as Wine FSR can be finnicky, and as you just saw, varies between versions. So we can keep it in the gamescope menu, and remove it otherwise.

**Comment by SuperSamus** [comment]
Rebased and addressed #3065.

**Comment by orowith2os** [comment]
Needs a quick look over from @mirkobrombin, otherwise I'd like to land this.

**Review by mirkobrombin — APPROVED** [review]
I checked the changes and looks goof to me. I am not able to test the changes by myself.

Thanks for your effort.

**Review by orowith2os — APPROVED** [review]

**Comment by orowith2os** [comment]
Will test later to make sure FSR works; then I'm merging, with some other fixes, for 51.10.

**Comment by koplo199** [comment]
After investigation, I **strongly** disagree with the proposal made here of only setting FSR inside gamescope.
Here is why:

1. Bottles doesn't explicitly disable FSR, it just don't force enable it if not instructed to, proof: [here](https://github.com/bottlesdevs/Bottles/blob/af3a7ccf44cd17d40160b1db66a5c8177b6f4cf5/bottles/backend/wine/executor.py#L80-L82) and [here](https://github.com/bottlesdevs/Bottles/blob/af3a7ccf44cd17d40160b1db66a5c8177b6f4cf5/bottles/backend/wine/executor.py#L80-L82) . So not only the bug doesn't exist in upstream anymore (see my last [comment](https://github.com/bottlesdevs/Bottles/pull/2840#issuecomment-1685212743)), but we weren't even affected by it in the first place. There's no reason to change anything here.
2. We loose the ability to easily set FSR without gamescope.
3. We loose program specific FSR setting. Now, it can only be enabled/disabled globally (and with gamescope only).

Only remains duplicated FSR setting issue, which is a 3 lines fixes at most.

**Comment by SuperSamus** [comment]
1. The goal of this PR wasn't to solve that. Assuming Wine-GE continued to enable FSR by default today, #2636's status wouldn't have changed at all. 
The intention is: by making the FSR setting associated with Gamescope, it is more obvious that Wine-GE still does whatever it wants with FSR.

2. It was also easy to have the setting enabled while it's doing absolutely nothing.
With how messy the `WINE_FULLSCREEN_FSR` situation is, making the user manage it manually with environment variables is more appropriate.
Not everything needs to be an explicit setting. For instance, why is DXVK's frame rate limit not a setting? Because `DXVK_FRAME_RATE` only applies to specific situations, so the user doesn't toggle a setting and get surprised when it does nothing.
IMO, `WINE_FULLSCREEN_FSR` is in a similar situation.

3. Not the biggest deal since it can be toggled with `Super+U`.
I do not disagree that it's a downgrade, but the option *needs* to be in a Gamescope's submenu (making Preference Override do this is out of scope for this PR).

**Comment by koplo199** [comment]
> 1. The goal of this PR wasn't to solve that. Assuming Wine-GE continued to enable FSR by default today, [[Bug]: FSR enabled in Wine-GE when disabled in Bottles #2636](https://github.com/bottlesdevs/Bottles/issues/2636)'s status wouldn't have changed at all.

It got disabled by default at the same time it was fixed, so that issue could be closed safely.

> It was also easy to have the setting enabled while it's doing absolutely nothing.

> With how messy the WINE_FULLSCREEN_FSR situation is making the user manage it manually with environment variables is more appropriate.

It is also easy to call the WINE_FULLSCREEN_FSR situation messy, when in reality this designation only rely on one upstream bug that got fixed in the meantime. Unless there's other bugs I'm unaware of?

> Not everything needs to be an explicit setting. For instance, why is DXVK's frame rate limit not a setting? Because DXVK_FRAME_RATE only applies to specific situations, so the user doesn't toggle a setting and get surprised when it does nothing.

I absolutely agree with this observation. I only disagree with the conclusion: it is fairly common to have a resolution setting/slider in-game. Having FSR kicking in while not on 100% resolution, basically providing better performance with minimal visual loss, is not as niche as `DXVK_FRAME_RATE`. I do not even see a point in setting `DXVK_FRAME_RATE`, maybe except for games with buggy VSYNC implementation.

> Not the biggest deal since it can be toggled with Super+U.

I invested a lot of time researching the topic and this is something I didn't even know. So, first: thanks for the tip :sweat_smile:, and second: I think I take no risk saying the vast majority of users may, just like I was 30 seconds ago, not be aware of the existence of this setting.

> I do not disagree that it's a downgrade, but the option needs to be in a Gamescope's submenu (making Preference Override do this is out of scope for this PR).

I don't understand why you say it _needs_ be in the gamescope's submenu: what is the problem with using the global/program FSR value to decide if FSR is enabled inside gamescope?

Also, something that hasn't been mentioned is that wine FSR is far from exclusive to Wine-GE. For instance the default soda runner also include these [patches](https://github.com/bottlesdevs/build-tools/blob/c0dae7673df897fd38d51c72741ea33e74e5b5fc/runners/vaniglia/wine-tkg-valve.cfg#L22). They are also enabled by default in upstream [wine-tkg](https://github.com/SingleInfiniteLoop/wine-tkg-git/blob/34a641ed97570d93db6410e2a74fa588cf1d2ee6/wine-tkg-git/customization.cfg#L45), meaning it will be available for any runner using tkg patches, assuming of course the said runner does not explicitly set `_use_GE_patches="false"`.

**Comment by TheEvilSkeleton** [comment]
I apologize for the long overdue review of this, as some of us were burned out and needed a break. Is it something you want to continue working on?

**Comment by TheEvilSkeleton** [comment]
Huh, it seems like GitHub doesn't notify me via emails when a push happens. I'm really sorry for being a week late and merging things that ended up conflicting with this MR :/

**Comment by TheEvilSkeleton** [comment]
@SuperSamus @koplo199 I'd like to revive the discussion in https://github.com/bottlesdevs/Bottles/pull/2840#issuecomment-1750397125. I don't really know which decision to go for, but I'm more in favor of accepting this MR, mainly because it cleans up a good chunk of the code. I think it's wise(r) to have the runner decide for the user whether FSR should be enabled or not.

**Comment by mirkobrombin** [comment]
I reviewed all the comments and I am fine with merging this MR. Can you please fix conflicts? Sorry for the delay, I was busy.

**Comment by SuperSamus** [comment]
Whoops, I missed the email.
Here it is.
